From a09576173d3e5fac9241534cdfc006a40fad9029 Mon Sep 17 00:00:00 2001
From: Paul Keith <javelinanddart@gmail.com>
Date: Fri, 22 Dec 2017 16:03:10 +0100
Subject: [PATCH 09/16] klte-common: Fixup RIL_Call structure

* Because of how structure alignment works,
  adding these unused chars here is unneeded
* This makes this structure more understandable
  when looking at the stock RIL java class
* Move Samsung call details as the 8 bytes before
  the RIL_UUS_Info pointer, to match stock

Change-Id: I2e62be0b1774209c0165ece90588ecb7aeb042e3
---
 include/telephony/ril.h | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/include/telephony/ril.h b/include/telephony/ril.h
index 357603c..a0b0ed6 100644
--- a/include/telephony/ril.h
+++ b/include/telephony/ril.h
@@ -422,8 +422,6 @@ typedef struct {
     RIL_CallState   state;
     char            index;      /* Connection Index for use with, eg, AT+CHLD */
     char            call_id;    /* Samsung call_id */
-    char            foo1;       /* Samsung */
-    char            foo2;       /* Samsung */
     int             toa;        /* type of address, eg 145 = intl */
     char            isMpty;     /* nonzero if is mpty call */
     char            isMT;       /* nonzero if call is mobile terminated */
@@ -433,11 +431,13 @@ typedef struct {
     char            isVoicePrivacy;     /* nonzero if CDMA voice privacy mode is active */
     char *          number;     /* Remote party number */
     int             numberPresentation; /* 0=Allowed, 1=Restricted, 2=Not Specified/Unknown 3=Payphone */
-    char            call_type;  /* Samsung*/
-    char            call_domain;/* Samsung*/
-    int             namePresentation; /* 0=Allowed, 1=Restricted, 2=Not Specified/Unknown 3=Payphone */
-    char *          csv;        /* Samsung */
     char *          name;       /* Remote party name */
+    int             namePresentation; /* 0=Allowed, 1=Restricted, 2=Not Specified/Unknown 3=Payphone */
+    union {
+       char         call_type;  /* Samsung call details */
+       char *       csv;        /* Samsung call details */
+    };
+    char            call_domain;/* Samsung call details */
     RIL_UUS_Info *  uusInfo;    /* NULL or Pointer to User-User Signaling Information */
 } RIL_Call;
 
-- 
2.7.4

